data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -75,4 +83,24 @@ public void process(ContentCacheUpdaterSource source, ContentCacheUpdaterSink si
         }
         return contentData.getParsedData(parsedData, newProperties);
     }
-}
+
+    public void updateDialCodeToCache(List<String> dialCodeList, int storeId, String apiUrl) {
+        if (null != dialCodeList) {
+            dialCodeList.forEach(dialCode -> {
+                if (!dialCode.isEmpty()) {
+                    String metaData = readFromCache(dialCode, storeId);
+                    if (null == metaData || metaData.isEmpty()) {
+                        LOGGER.info("", String.format("Invoking DialCode API to fetch the metadata for this dialCode( %s )", dialCode));
+                        Object dialCodeMetadata = contentData.getMetadata(apiUrl.concat(dialCode), this.restUtil, contentCacheConfig.getAuthorizationKey(), "dialcode");
+                        if (null != dialCodeMetadata) {
+                            LOGGER.info("", String.format("Inserting dialCode( %s ) metadata to cache system", dialCode));
+                            addToCache(dialCode, gson.toJson(dialCodeMetadata), storeId);
+                        } else {
+                            LOGGER.info("", String.format("API did not fetched any dialCode metadata!!! %s", dialCode));
```

Comment:
Can we capture some metrics on - how many dialcodes came, how many are available in cache and how many of them have been fetched from the API